### PR TITLE
Smart on fhir docs

### PR DESCRIPTION
**Pull request by himeshsiriwardana**
screencapture-localhost-8000-asgardeo-docs-tutorials-smart-on-fhir-2024-09-26-19_01_33

### CODE REVIEW

File: en/asgardeo/docs/tutorials/smart-on-fhir.md [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+
+Follow the tutorial below to learn about how you can leverage Asgardeo to implement a robust authentication and authorization mechanism for your healthcare applications built on SMART on FHIR.
+
+## What is FHIR?
+
+[FHIR](https://hl7.org/fhir/R4/index.html){target="_blank"} (Fast Healthcare Interoperability Resources) is a standard introduced by HL7 for securely exchanging healthcare data. The standard defines a set of structured data formats for key healthcare resources such as patients, practitioners, diagnostics and medications. This ensures that regardless of how the data are stored in one system, they can be understood and processed by another system even if it is running on a different technology.
+
+## What is SMART on FHIR?
+
+[SMART ON FHIR](https://hl7.org/fhir/smart-app-launch/index.html){target="_blank"} is a framework that extends FHIR and allows third-party developers to create applications that seamlessly integrate with Electronic Health Records (EHR) systems. There are two main advantages of using SMART ON FHIRE for building healthcare apps:
```
Comment (isuruuy429) [inline en/asgardeo/docs/tutorials/smart-on-fhir.md:11]: SMART on FHIR

File: en/asgardeo/docs/tutorials/smart-on-fhir.md [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+
+3. Do the following for all the REST APIs listed on the above table.
+
+    1. Click on **Authorize an API Resource** and do the following:
+    
+    2. Select the resource from the **API Resource** dropdown.
+
+        !!! note
+
+            Be sure to select the API resources listed under **Management API**.
```
Comment (isuruuy429) [inline en/asgardeo/docs/tutorials/smart-on-fhir.md:150]: This note is not needed. The example was provided to enable the scopes for 'API Resource Management API’.

File: en/asgardeo/docs/tutorials/smart-on-fhir.md [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+
+- Get a bearer token to access Asgardeo REST APIs
+
+- Create the `fhirUser` user attribute. This will be used as the identity of a user accessing FHIR resources.
+
+- Create the `fhirUser` scope. Applications can request this scope to access the `fhirUser` attribute of a user.
+
+- Add the relevant OIDC and SCIM dialects of the `fhirUser` user attribute.
+
+- Update the `fhirUser` attribute of a selected user to `Patient/1`.
```
Comment (isuruuy429) [inline en/asgardeo/docs/tutorials/smart-on-fhir.md:172]: Update the `fhirUser` attribute of a selected user to PatientId.

File: en/asgardeo/docs/tutorials/smart-on-fhir.md [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+
+Follow the steps below to download and run the Postman collection:
+
+1. Download the Postman collection from [Github](https://github.com/wso2-enterprise/open-healthcare/blob/main/scripts/postman-collections/smartonfhir-asgardeo.postman_collection.json){target="_blank"} and import it to Postman.
+
+2. In the Postman collection named **Asgardeo**, go to the **Authorization** tab and do the following:
+
+    1. Under **Auth Type**, select **Basic Auth**.
+
+    2. Enter the client ID and client secret of your application as username and password respectively.
```
Comment (isuruuy429) [inline en/asgardeo/docs/tutorials/smart-on-fhir.md:186]: Shall we make the client id and client secret values not visible?

Comment (isuruuy429) [reply en/asgardeo/docs/tutorials/smart-on-fhir.md:186]: Also shall we provide a screenshot of the application created in Asgardeo after running the Postman Collection?

File: en/asgardeo/docs/tutorials/smart-on-fhir.md [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+
+    3. Ensure no errors occur and troubleshoot any errors that arise.
+
+        ![Postman test cases]({{base_path}}/assets/img/tutorials/smart-on-fhir/fhir-postman-test-cases.png)
+
+### Step 4: Verify that everything works
+
+Now that you have created the `fhirUser` user attribute and registered your SMART on FHIR app in Asgardeo, let's try to log into your application as the user and obtain the ID token. If everything has worked, the ID token should contain the `fhirUser` attribute.
+
+To do so,
```
Comment (isuruuy429) [inline en/asgardeo/docs/tutorials/smart-on-fhir.md:222]: Shall we add the screenshot of retrieving the id-token? (The 'get-token' request's Authorization tab)

File: en/asgardeo/docs/tutorials/smart-on-fhir.md (hunk 3 above)
Comment (isuruuy429) [reply en/asgardeo/docs/tutorials/smart-on-fhir.md:172]: It needs to add the following steps as well before the last step. 
'Create API Resources policy for SMART App Launch'

File: en/asgardeo/docs/tutorials/smart-on-fhir.md (hunk 2 above)
Comment (himeshsiriwardana) [reply en/asgardeo/docs/tutorials/smart-on-fhir.md:150]: Will reword

File: en/asgardeo/docs/tutorials/smart-on-fhir.md (hunk 3 above)
Comment (himeshsiriwardana) [reply en/asgardeo/docs/tutorials/smart-on-fhir.md:172]: Will change to patientId (Patient/1)

File: en/asgardeo/docs/tutorials/smart-on-fhir.md (hunk 4 above)
Comment (himeshsiriwardana) [reply en/asgardeo/docs/tutorials/smart-on-fhir.md:186]: Will add the SS of the API authorization tab

**Comment by isuruuy429** [comment]
Have we added the following 2 APIs to the API list?
- SCIM2 Roles API
- SCIM2 Bulk API

**Comment by himeshsiriwardana** [comment]
Updated documentation:

screencapture-localhost-8000-asgardeo-docs-tutorials-smart-on-fhir-2024-10-29-14_07_39

**Review by isuruuy429 — APPROVED** [review]

**Review by pavinduLakshan — APPROVED** [review]

**Pull request merged**